server/src/main/java/de/uftos/services/StudentService.java

```diff
@@ -39,7 +40,28 @@ public StudentService(StudentRepository repository) {
    */
   public Page<Student> get(Pageable pageable, Optional<String> firstName,
                            Optional<String> lastName, Optional<String[]> tags) {
-    return this.repository.findAll(pageable);
+    Specification<Student> spec = createSpecification(firstName, lastName, tags);
+    return this.repository.findAll(spec, pageable);
+  }
+
+  private Specification<Student> createSpecification(Optional<String> firstName,
+                                                         Optional<String> lastName,
+                                                         Optional<String[]> tags) {
+    Specification<Student> spec = Specification.where(null);
```

Comment:
null as a parameter, might be dangerous, was this tested?
Using cb.disjunction() as an always false might be a better option.

Reply:
Yes, it was tested. where(null) is also always false.